### PR TITLE
Adds naive-caching to demo Header and status manipulation and test.

### PR DESCRIPTION
**Pull request by ignasi35**

### CODE REVIEW

**Comment by ignasi35** [comment]
This is a one-shot PR, there are several commits but it should be squashed.

File: naive-http-caching/naive-http-caching-java-sbt/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Header Manipulation and HTTP testing
```
Comment (TimMoore) [inline naive-http-caching/naive-http-caching-java-sbt/README.md]: If the purpose of the recipe is to demonstrate header manipulation, I think the name of the directory/project should reflect that. If someone is looking for a generic header manipulation example and are not doing caching specifically, they might not think to look in something called "naive-http-caching".

File: naive-http-caching/naive-http-caching-java-sbt/hello-impl/src/main/resources/application.conf [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# 
+#
+play.crypto.secret=whatever
+play.modules.enabled += com.example.hello.impl.HelloModule
+
+hello.cassandra.keyspace = hello
+
+cassandra-journal.keyspace = ${hello.cassandra.keyspace}
+cassandra-snapshot-store.keyspace = ${hello.cassandra.keyspace}
+lagom.persistence.read-side.cassandra.keyspace = ${hello.cassandra.keyspace}
```
Comment (TimMoore) [inline naive-http-caching/naive-http-caching-java-sbt/hello-impl/src/main/resources/application.conf]: You could trim out all of the unused Cassandra config

File: naive-http-caching/naive-http-caching-java-sbt/README.md (hunk 1 above)
Comment (renatocaval) [reply naive-http-caching/naive-http-caching-java-sbt/README.md]: @TimMoore I think that the initial goal was not to demonstrate generic header manipulation, but indeed that is maybe a better option. 

Instead, we could have different, but very simple, ServiceCalls each demonstrating a different header manipulation use case. The If-None-Match and ETag would one of the possible use cases.

Comment (ignasi35) [reply naive-http-caching/naive-http-caching-java-sbt/README.md]: I didn't know how to name the folders. The purpose of the recipe wasn't clear in my head when I started. :-(

Comment (ignasi35) [reply naive-http-caching/naive-http-caching-java-sbt/README.md]: @renatocaval I like the idea. I'm only worried too many examples could make the sample too hard to follow. I think we'll find out if/when we add more calls.

**Comment by ignasi35** [comment]
This is ready for a second pass and squash+merge.
